src/Utilities/Compiler/RuleLevel.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#if NET_ANALYZERS || FXCOP_ANALYZERS
+
+namespace Microsoft.CodeAnalysis
+{
+    internal enum RuleLevel
```

Comment:
@sharwell @genlu You may want to start the review from this file. This closely matches the rule levels that we decided. I have excluded the SDL required rule level and made the security rules use `BuildWarning`, which is basically level 2 in the spreadsheet.

`IdeSuggestion` and `IdeHidden_BulkConfigurable` are basically rule level 3 in the spreadsheet, with the former being no false positive/useful as IDE suggestion case and latter being potential false positive case.

`Disabled` and `CandidateForRemoval` are levels 4 and 5 respectively, as per the spreadsheet.